db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/StringAggregator.java

```diff
@@ -0,0 +1,167 @@
+import com.splicemachine.db.iapi.types.DataTypeDescriptor;
+import com.splicemachine.db.iapi.types.DataValueDescriptor;
+import com.splicemachine.db.iapi.types.NumberDataValue;
+import com.splicemachine.db.iapi.types.SQLVarchar;
+import com.splicemachine.db.iapi.types.StringDataValue;
+import com.splicemachine.db.iapi.types.TypeId;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+public final class StringAggregator extends OrderableAggregator {
+    private StringBuilder aggregator;
+    private String separator;
+
+    public StringAggregator(){}
+
+    public StringAggregator(String separator) {
+        this.separator = separator;
+    }
+
+    public Class getAggregatorClass() {
+        return aggregator.getClass();
+    }
+
+    @Override
+    public ExecAggregator setup(ClassFactory cf, String aggregateName, DataTypeDescriptor returnDataType, DataValueDescriptor param) throws StandardException {
+        if (aggregator == null) {
+            this.aggregator = new StringBuilder();
+            separator = param.getString();
+        }
+        return this;
+    }
+
+    protected void accumulate(DataValueDescriptor addend)
+            throws StandardException {
+        if (aggregator.length() == 0)
+            aggregator.append(addend.getString());
+        else
+            aggregator.append(separator).append(addend.getString());
+        return;
+
+    }
+
+    public void merge(ExecAggregator addend)
+            throws StandardException {
+        if (addend == null) return; //ignore null entries
+        StringBuilder other = ((StringAggregator) addend).aggregator;
+        if (aggregator.length() == 0)
+            aggregator.append(other);
+        else
+            aggregator.append(separator).append(other);
+    }
+
+    public void add(DataValueDescriptor addend) throws StandardException {
+        accumulate(addend);
+    }
+
+    /**
+     * Return the result of the aggregation.  If the count
+     * is zero, then we haven't averaged anything yet, so
+     * we return null.  Otherwise, return the running
+     * average as a double.
+     *
+     * @return null or the average as Double
+     */
+    public DataValueDescriptor getResult() throws StandardException {
+        if (this.aggregator.length() == 0) {
+            return null;
+        }
+        return new SQLVarchar(aggregator.toString());
+    }
+
+    @Override
+    public ExecAggregator newAggregator() {
+        StringAggregator strAggregator = new StringAggregator(separator);
+        strAggregator.aggregator = new StringBuilder();
+        return strAggregator;
+    }
+
+    /////////////////////////////////////////////////////////////
+    //
+    // EXTERNALIZABLE INTERFACE
+    //
+    /////////////////////////////////////////////////////////////
+
+    /**
+     * @throws IOException on error
+     */
+    public void writeExternal(ObjectOutput out) throws IOException {
+        out.writeObject(aggregator);
```

Comment:
Please consider writing a class version number (e.g. 1) as part of SerDe in case this is every stored on disk.